ui-tests/cypress/integration/components/ux_fragments.spec.js

```diff
@@ -1,808 +1,665 @@
+            page.getContent().getPagination().getItemsCurrent().should('have.text', '1-1');
+            page.getContent().getPagination().getItemsTotal().should('have.text', '1');
+          });
+    });
+
+    it([Tag.FEATURE, 'ENG-3522'], 'Search with widget filter', () => {
+      cy.fixture('data/uxFragmentsWidgetTypes.json').then(widgetTypes => getArrayRandomElement(widgetTypes)).then(widgetType =>
+          openUXFragmentsPage()
+              .then(page => page.getContent().getWidgetFilter().then(select => page.getContent().select(select, widgetType.name)))
+              .then(page => page.getContent().clickSearchSubmitButton())
               .then(page =>
-                  page.getContent().getSearchCodeInput().then(input => page.getContent().type(input, fragment.code.substring(3, 7))))
+                  page.getContent().getTableRows()
+                      .each(row => cy.get(row).children(htmlElements.td).eq(2).should('have.text', widgetType.name))
+                      .then(rows => {
+                        page.getContent().getPagination().getItemsCurrent().invoke('text').should('be.equal', `1-${Math.min(rows.length, 10)}`);
+                        page.getContent().getPagination().getItemsTotal().invoke('text').should('be.equal', `${rows.length}`);
+                      })));
+    });
+
+    it([Tag.FEATURE, 'ENG-3522'], 'Search with plugin filter', () => {
+      cy.fixture('data/uxFragmentsPlugins.json').then(plugins => getArrayRandomElement(plugins)).then(plugin =>
+          openUXFragmentsPage()
+              .then(page => page.getContent().getPluginFilter().then(select => page.getContent().select(select, plugin.name)))
               .then(page => page.getContent().clickSearchSubmitButton())
               .then(page =>
-                  page.getContent().getTableRows().should('have.length', 1)
-                      .each(row => cy.get(row).children(htmlElements.td).eq(0).should('contain', fragment.code.substring(3, 7))));
-          cy.get('@currentPage')
-            .then(page => {
-              page.getContent().getPagination().getItemsCurrent().invoke('text').should('be.equal', '1-1');
-              page.getContent().getPagination().getItemsTotal().invoke('text').should('be.equal', '1');
-            });
-        });
-      });
+                  page.getContent().getTableRows()
+                      .each(row => cy.get(row).children(htmlElements.td).eq(4).should('have.text', plugin.name))
+                      .then(rows => {
+                        page.getContent().getPagination().getItemsCurrent().invoke('text').should('be.equal', `1-${Math.min(rows.length, 10)}`);
+                        page.getContent().getPagination().getItemsTotal().invoke('text').should('be.equal', `${rows.length}`);
+                      })));
+    });
 
-      it([Tag.FEATURE, 'ENG-3522'], 'Bug: Widget type does not filter', () => {
```

Comment:
Do not _hard-code_ temporary information! the bug will be fixed and there's no reason to have to update the test